solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -442,6 +482,51 @@ pub struct UpdateAdmin<'info> {
     pub admin_authority: Signer<'info>,
 }
 
+#[derive(Accounts)]
+#[instruction(base: Pubkey, handle_seed: [u8;16], user_bump:u8)]
+pub struct InitAppDelegate<'info> {
+    #[account()]
+    pub admin: Account<'info, AudiusAdmin>,
+    #[account(seeds = [&base.to_bytes()[..32], handle_seed.as_ref()], bump = user_bump)]
+    pub app_user: Account<'info, User>,
+    #[account(
+        init,
+        payer = payer,
+        seeds = [b"app_delegate", app_user.key().as_ref()],
+        bump,
+        space = APP_DELEGATE_ACCOUNT_SIZE
+    )]
+    pub app_delegate_pda: Account<'info, AppDelegate>,
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+#[instruction(base: Pubkey, handle_seed: [u8;16], user_bump:u8, user_authority_delegate: Pubkey)] // TODO use UserHandle
+pub struct RevokeAppDelegate<'info> {
+    #[account()]
+    pub admin: Account<'info, AudiusAdmin>,
+    #[account(
+        seeds = [&base.to_bytes()[..32], handle_seed.as_ref()],
```

Comment:
needs the same seed structure here - but not sure why we need user handle stuff?

Reply:
oh nvm this is for the User acct